Description: Add Linux 2.6.38 support
 The following commit renamed acquire_console_sem to console_lock and
 release_console_sem to console_unlock on the Linux kernel:
 http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=ac751efa6a0d70f2c9daef5c7e3a92270f5c2dff
Author: Miguel Colon <debian.micove@gmail.com>
Forwarded: no
Last-Update: 2011-02-01

--- fglrx-driver-11-1.orig/common/lib/modules/fglrx/build_mod/firegl_public.c
+++ fglrx-driver-11-1/common/lib/modules/fglrx/build_mod/firegl_public.c
@@ -850,7 +850,12 @@ static int fglrx_pci_suspend(struct pci_
      * happen much less frequent then without this workaround.
      */
     if (state == PM_EVENT_SUSPEND)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+        console_lock();
+#else
         acquire_console_sem();
+#endif
+
 
     if (firegl_cail_powerdown(privdev, state))
         ret = -EIO;
@@ -872,7 +877,11 @@ static int fglrx_pci_suspend(struct pci_
     }
 
     if (state == PM_EVENT_SUSPEND)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+        console_unlock();
+#else
         release_console_sem();
+#endif
 
     KCL_DEBUG_TRACEOUT(FN_FIREGL_ACPI, ret, NULL);  
     
@@ -896,7 +905,11 @@ static int fglrx_pci_resume(struct pci_d
     if (PMSG_EVENT(pdev->dev.power.power_state) == 0) return 0;
 
     if (PMSG_EVENT(pdev->dev.power.power_state) == PM_EVENT_SUSPEND)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+        console_lock();
+#else
         acquire_console_sem();
+#endif
 
 #ifdef FIREGL_USWC_SUPPORT
     // Restore the PAT after resuming from S3 or S4.
@@ -921,7 +934,11 @@ static int fglrx_pci_resume(struct pci_d
     firegl_cail_powerup(privdev);
 
     if (PMSG_EVENT(pdev->dev.power.power_state) == PM_EVENT_SUSPEND)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+        console_unlock();
+#else
         release_console_sem();
+#endif
 
     PMSG_EVENT(pdev->dev.power.power_state) = 0;
     KCL_DEBUG_TRACEOUT(FN_FIREGL_ACPI, 0, NULL);  
